testing/integration/otel_helm_test.go

```diff
@@ -99,10 +101,8 @@ func TestOtelKubeStackHelm(t *testing.T) {
 
 				// override secrets reference with env variables
 				JSONValues: []string{
-					fmt.Sprintf(`collectors.cluster.env[1]={"name":"ELASTIC_ENDPOINT","value":"%s"}`, kCtx.esHost),
-					fmt.Sprintf(`collectors.cluster.env[2]={"name":"ELASTIC_API_KEY","value":"%s"}`, kCtx.esAPIKey),
-					fmt.Sprintf(`collectors.daemon.env[2]={"name":"ELASTIC_ENDPOINT","value":"%s"}`, kCtx.esHost),
-					fmt.Sprintf(`collectors.daemon.env[3]={"name":"ELASTIC_API_KEY","value":"%s"}`, kCtx.esAPIKey),
+					fmt.Sprintf(`collectors.gateway.env[1]={"name":"ELASTIC_ENDPOINT","value":"%s"}`, kCtx.esHost),
```

Comment:
I am sorry for the request but how hard is it to align the code of this test with [these one](https://github.com/elastic/elastic-agent/blob/main/testing/integration/kubernetes_agent_standalone_test.go#L267). Feel free to say no if it too much of an ask, we can do that in a follow-up PR 🙂

Reply:
refactored during https://github.com/elastic/elastic-agent/pull/6444/commits/c5c52aa53918fc9d074798a738a888303a4900af